metadata_mapper/mappers/oai/contentdm_mapper.py

```diff
@@ -0,0 +1,133 @@
+            "language": self.split_and_flatten('language'),
+            "subject": self.map_subject()
+        }
+
+    @staticmethod
+    def identifier_match():
+        return ["cdm/ref"]
+
+    def map_is_shown_at(self):
+        return self.get_matching_identifier(last=True)
+
+    def map_is_shown_by(self):
+        """
+        This was originally a "post-map" function. but really, this is seems like it should be map_is_shown_by()
+
+        Comment from calisphere function:
+        To run post mapping. For this one, is_shown_by needs sourceResource/type
+        """
+        record_type = self.map_type()
+        if not record_type:
+            return
+
+        record_type_list = [record_type] if isinstance(record_type, str) else record_type
+
+        if "sound" in [t.lower() for t in record_type_list]:
+            return None
+
+        return self.get_preview_image_url()
+
+    def map_type(self):
+        """Used by map_is_shown_by() in this class, so cannot be directly added to the mapping"""
+        return self.split_and_flatten('type')
+
+    def map_spatial(self):
+        values = [v for v in self.collate_fields(["coverage", "spatial"]) if v]
+        if not values:
+            return
+
+        split_values = [c.split(';') for c in filter(None, values)]
+
+        return list([s.strip() for s in itertools.chain.from_iterable(split_values)])
+
+    def map_subject(self):
+        subject = self.source_metadata.get('subject')
+        if not subject:
+            return
+
+        return [{'name': v} for v in self.split_and_flatten('subject')]
+
+    def get_preview_image_url(self):
+        """
+        see get_larger_preview_image(), below, as it is mapped to isShownBy as well
+        """
+        larger_preview_image = self.get_larger_preview_image_url()
+        if larger_preview_image:
+            return larger_preview_image
+
+        parts = self.get_identifier_parts()
+        return '/'.join((parts["base_url"], 'utils', 'getthumbnail',
+                         'collection', parts["collection_id"], 'id', parts["object_id"]))
+
+    def get_larger_preview_image_url(self):
+        image_info = self.get_image_info()
+        if not image_info['height'] > 0:
+            return
+
+        max_dim = 1024.0
+        if image_info['height'] >= image_info['width']:
+            scale = int((max_dim / image_info['height']) * 100)
+        else:
+            scale = int((max_dim / image_info['width']) * 100)
+        scale = 100 if scale > 100 else scale
+        return '{}&action=2&DMHEIGHT=2000&DMWIDTH=2000&DMSCALE={}'.format(self.get_url_image_info(), scale)
+
+    def get_image_info(self):
+        """
+        TODO: address HTTP request taking place in here
+        """
+        image_info = {'height': 0, 'width': 0}
+        identifier = self.get_matching_identifier()
+        if not identifier:
+            return image_info
+
+        response = requests.get(self.get_url_image_info()).json()['imageinfo']
```

Comment:
Add a `TODO: alternative strategy for making another request` for @barbarahui and I to pick up later. Since these mappers will be running in parallel in the AWS Lambda environment, this will hammer the contributor's system and is probably not ideal.